src/pages/index.js

```diff
@@ -81,35 +143,174 @@ profileEditButton.addEventListener("click", () => {
+//         editAvatarModal.close();
+//       })
+//       .catch((err) => console.error("Error updating avatar:", err))
+//       .finally(() => {
+//         editAvatarModal.setLoadingState(false);
+//       });
+//   } else {
+//     console.error("Avatar Link is not defined");
+//   }
+// }
+
+function handleAvatarEditSubmit(input) {
+  const link = input.avatar;
+
+  if (link) {
+    // set loading state to true
+    editAvatarModal.setLoadingState(true);
+    // the right api call is updateAvatar
+    api
+      .updateAvatar(link)
+      .then((userData) => {
+        // set avatar info
+        userInfo.setAvatarInfo(userData.avatar);
+
+        editAvatarModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        // set loading state to false
+        editAvatarModal.setLoadingState(false);
+      });
+  } else {
+    console.error("Avatar Link is not defined");
+  }
+}
+
 function handleProfileEditSubmit(formValues) {
   userInfo.setUserInfo({
     name: formValues.title,
     about: formValues.card__description,
   });
   editProfileModal.close();
 }
-function handleAddCardFormSubmit(formValues) {
-  const name = formValues.title;
-  const link = formValues.url;
-
-  const card = createCard({ name, link });
-  cardList.addItem(card);
-  addCardFormElement.reset();
-  addCardModal.close();
+// function handleAddCardFormSubmit(formValues) {
+//   const name = formValues.title;
+//   const link = formValues.url;
+
+//   const card = createCard({ name, link });
+//   cardList.addItem(card);
+//   addCardFormElement.reset();
+//   addCardModal.close();
+// }
+
+function handleDeleteClick(card) {
+  const cardId = card._id;
+  if (!cardId) {
+    console.error("card._id is undefined");
+    return;
+  }
+  confirmDeleteModal.setSubmitFunction(() => {
+    api
+      .deleteCard(cardId)
+      .then((res) => {
+        console.log(res);
+        card._handleDeleteCard();
+        confirmDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  confirmDeleteModal.open();
 }
 
-function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardPreview,
+    (data) => {
+      if (!data.isLiked) {
+        api
+          .likeCard(data._id, { method: "PUT" })
+          .then((res) => {
+            console.log(res);
+            data.isLiked = true;
+            card.toggleLike();
+          })
+          .catch((err) => {
+            console.error(err);
+          });
+      } else {
+        api
+          .dislikeCard(data._id, { method: "DELETE" })
+          .then((res) => {
+            console.log(res);
+            data.isLiked = false;
+            card.toggleLike();
+          })
+          .catch((err) => {
+            console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors